File: bench/cases/generic-id-1m/main.go
package main

import "fmt"

func id[T any](x T) T {
	return x
}

func loopSum(xs []int64) int64 {
	var sum int64 = 0
	for _, x := range xs {
		sum = sum + id(x)
	}
	return sum
}

func main() {
	var xs []int64
	for i := int64(0); i < 10000000; i++ {
		xs = append(xs, i)
	}
	fmt.Println(loopSum(xs))
}

File: bench/cases/generic-id-1m/main.ts
function id<T>(x: T): T {
  return x
}

function loopSum(xs: number[]): number {
  let sum = 0
  for (let i = 0; i < xs.length; i = i + 1) {
    sum = sum + id(xs[i])
  }
  return sum
}

const xs: number[] = []
for (let i = 0; i < 10000000; i = i + 1) {
  xs.push(i)
}
console.log(loopSum(xs))
